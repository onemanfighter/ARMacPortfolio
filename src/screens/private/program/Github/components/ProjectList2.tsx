import { GithubState, appStore, githubSelector, useShallow } from '@appStore';
import { Box, Button } from '@chakra-ui/react';

const projectList = [
  {
    name: 'RaikwarAmit',
    project: 'raikwaramit',
  },
  {
    name: 'CryptoAtlas',
    project: 'CryptoAtlas',
  },
  {
    name: 'Courses',
    project: 'courses',
  },
  {
    name: 'Sorting visualizer',
    project: 'sorting_visualizer',
  },
  {
    name: 'Trackify',
    project: 'Trackify',
  },
  {
    name: 'React Markdown Demo',
    project: 'ReactMarkdownDemo',
  },
  {
    name: 'Authentication Firebase',
    project: 'AuthenticationFirebase',
  },
  {
    name: 'Some React Concepts',
    project: 'SomeReactConcepts',
  },
  {
    name: 'React Hooks',
    project: 'react_hooks',
  },
  {
    name: 'Room Database Module',
    project: 'RoomDatabaseModule',
  },
  {
    name: 'Location demo app',
    project: 'Location_demo_app',
  },
  {
    name: 'DSA Practice',
    project: 'DSA-Practice',
  },
  {
    name: 'Mini quiz app',
    project: 'Mini-quiz-app',
  },
];

const ProjectList2 = () => {
  const { setProject } = appStore(useShallow(githubSelector));

  const handleProject = (project: string) => {
    setProject(project as GithubState['project']);
  };
  return (
    <Box
      display={'flow'}
      flexDir={'column'}
      style={{ overflowY: 'scroll', height: '32vh' }}
    >
      {projectList.map((data) => (
        <Button
          padding={2}
          margin={1}
          key={data.name}
          onClick={() => {
            handleProject(data.project);
          }}
        >
          {data.name}
        </Button>
      ))}
    </Box>
  );
};

export default ProjectList2;
