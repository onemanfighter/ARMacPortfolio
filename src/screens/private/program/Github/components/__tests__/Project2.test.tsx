import { fireEvent, render, screen } from '@testing-library/react';
import ProjectList2 from '../ProjectList2';

describe('ProjectList2', () => {
  it('should return default ProjectList1', () => {
    const { container } = render(<ProjectList2 />);

    expect(container).toMatchSnapshot();
  });

  it('should invoke callback on button press', () => {
    const { container } = render(<ProjectList2 />);

    fireEvent.click(screen.getByText('CryptoAtlas'));

    expect(container).toMatchSnapshot();
  });
});
