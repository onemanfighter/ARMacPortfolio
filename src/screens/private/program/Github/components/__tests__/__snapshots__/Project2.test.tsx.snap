// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList2 should invoke callback on button press 1`] = `
<div>
  <div
    class="css-19m134u"
    style="overflow-y: scroll; height: 32vh;"
  >
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      RaikwarAmit
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      CryptoAtlas
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Courses
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Sorting visualizer
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Trackify
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      React Markdown Demo
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Authentication Firebase
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Some React Concepts
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      React Hooks
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Room Database Module
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Location demo app
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      DSA Practice
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Mini quiz app
    </button>
  </div>
</div>
`;

exports[`ProjectList2 should return default ProjectList1 1`] = `
<div>
  <div
    class="css-19m134u"
    style="overflow-y: scroll; height: 32vh;"
  >
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      RaikwarAmit
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      CryptoAtlas
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Courses
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Sorting visualizer
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Trackify
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      React Markdown Demo
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Authentication Firebase
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Some React Concepts
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      React Hooks
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Room Database Module
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Location demo app
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      DSA Practice
    </button>
    <button
      class="chakra-button css-wgfsyt"
      type="button"
    >
      Mini quiz app
    </button>
  </div>
</div>
`;
